dff/messengers/telegram/abstract.py

```diff
@@ -0,0 +1,364 @@
+"""
+Interface
+------------
+This module implements various interfaces for :py:class:`~dff.messengers.telegram.messenger.TelegramMessenger`
+that can be used to interact with the Telegram API.
+"""
+
+from pathlib import Path
+from tempfile import gettempdir
+from typing import Callable, Optional, Sequence
+from pydantic import FilePath
+
+from telegram import (
+    InlineKeyboardButton,
+    InlineKeyboardMarkup,
+    InputMediaAnimation,
+    InputMediaAudio,
+    InputMediaDocument,
+    InputMediaPhoto,
+    InputMediaVideo,
+    Update,
+    Message as TelegramMessage,
+)
+from telegram.ext import Application, ExtBot, MessageHandler, CallbackQueryHandler, ContextTypes
+from telegram.ext.filters import ALL
+
+from dff.messengers.common import MessengerInterface
+from dff.pipeline.types import PipelineRunnerFunction
+from dff.script.core.message import (
+    Animation,
+    Audio,
+    Button,
+    CallbackQuery,
+    Contact,
+    DataAttachment,
+    Document,
+    Image,
+    Invoice,
+    Keyboard,
+    Location,
+    Message,
+    Poll,
+    PollOption,
+    Video,
+)
+
+
+class _AbstractTelegramInterface(MessengerInterface):  # pragma: no cover
+    request_attachments = {Location, Contact, Invoice, Poll, Audio, Video, Animation, Image, Document}
+    response_attachments = {Location, Contact, Poll, Audio, Video, Animation, Image, Document, Keyboard}
+
+    def __init__(self, token: str) -> None:
+        self.application = Application.builder().token(token).build()
+        self.application.add_handler(MessageHandler(ALL, self.on_message))
+        self.application.add_handler(CallbackQueryHandler(self.on_callback))
+
+    async def populate_attachment(self, attachment: DataAttachment) -> None:  # pragma: no cover
+        if attachment.title is not None and attachment.id is not None:
+            file_name = Path(gettempdir()) / str(attachment.title)
```

Comment:
Add option to pass a directory for attachments in the `__init__` method.

Also produce a warning if `tempdir` is used that the attachments may be removed along with the temp dir and `attachment.source` would be invalidated.